ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/manager/ForkIdManager.java

```diff
@@ -40,23 +41,25 @@
   private final List<ForkId> forkAndHashList;
 
   private final List<Predicate<ForkId>> forkIDCheckers;
+  private final List<Long> forks;
+  private final LongSupplier chainHeadSupplier;
+  private long lastHead;
+  private ForkId lastComputedForkId;
 
   public ForkIdManager(final Blockchain blockchain, final List<Long> forks) {
     checkNotNull(blockchain);
     checkNotNull(forks);
+    this.chainHeadSupplier = blockchain::getChainHeadBlockNumber;
     this.genesisHash = blockchain.getGenesisBlock().getHash();
     this.forkAndHashList = new ArrayList<>();
+    this.forks =
+        forks.stream()
+            .filter(fork -> fork > 0)
+            .distinct()
+            .sorted()
+            .collect(Collectors.toUnmodifiableList());
```

Comment:
Since `forks` doesn't include 0, `onlyZerosForkBlocks` is not doing what it looks like it's doing. As it happens, it will be true because `allMatch` is true for empty but it's pretty misleading right now.

Reply:
Fixed.